src/saveload/industry_sl.cpp

```diff
@@ -228,6 +230,20 @@ struct INDYChunkHandler : ChunkHandler {
 			} else if (IsSavegameVersionBefore(SLV_INDUSTRY_CARGO_REORGANISE)) {
 				LoadMoveAcceptsProduced(i, INDUSTRY_NUM_INPUTS, INDUSTRY_NUM_OUTPUTS);
 			}
+
+			if (IsSavegameVersionBefore(SLV_INDUSTRY_NUM_VALID_HISTORY)) {
+				/* Before this version we did not keep track of valid history, so assume it from the oldest non-zero value. */
+				uint oldest_valid = 0;
+				for (const auto &p : i->produced) {
+					if (!IsValidCargoType(p.cargo)) continue;
+					for (uint n = LAST_MONTH; n < static_cast<uint>(std::size(p.history)); ++n) {
+						if (p.history[n].production == 0 && p.history[n].transported == 0) continue;
+						oldest_valid = std::max(oldest_valid, n);
+					}
+				}
+				for (uint n = LAST_MONTH; n <= oldest_valid; ++n) SetBit(i->valid_history, n);
```

Comment:
Why `uint` instead of `size_t`? `std::size` returns `size_t` and all the `operator[]` calls require `size_t`, so a lot of conversions between `uint` and `size_t` need to be made here.

Reply:
And now with `size_t` MSVC moans :)